wdl/MainVcfQc.wdl

```diff
@@ -0,0 +1,843 @@
+    bootDiskSizeGb: select_first([runtime_override.boot_disk_gb, runtime_default.boot_disk_gb])
+  }
+
+  command <<<
+    set -eu -o pipefail
+    
+    # Clean fam file
+    /opt/sv-pipeline/scripts/vcf_qc/cleanFamFile.sh \
+      ~{samples_list} \
+      ~{ped_file} \
+      cleaned.fam
+    rm ~{ped_file} ~{samples_list}
+
+    # Only run if any families remain after cleaning
+    n_fams=$( grep -Ev "^#" cleaned.fam | wc -l )
+    echo -e "DETECTED $n_fams FAMILIES"
+    if [ $n_fams -gt 0 ]; then
+
+      # Make per-sample directory
+      mkdir ~{prefix}_perSample/
+
+      # Untar per-sample VID lists
+      mkdir tmp_untar/
+      tar -xvzf ~{per_sample_tarball} \
+        --directory tmp_untar/
+      for FILE in $( find tmp_untar/ -name "*.VIDs_genotypes.txt.gz" ); do
+        mv -v $FILE ~{prefix}_perSample/
+      done
+
+      # Subset fam file, if optioned
+      n_trios=$( grep -Ev "^#" cleaned.fam \
+                 | awk '{ if ($2 != "0" && $2 != "." && \
+                              $3 != "0" && $3 != "." && \
+                              $4 != "0" && $4 != ".") print $0 }' \
+                 | wc -l )
+      echo -e "DETECTED $n_trios COMPLETE TRIOS"
+      if [ $n_trios -gt ~{max_trios} ]; then
+        grep -E '^#' cleaned.fam > fam_header.txt
+        grep -Ev "^#" cleaned.fam \
+        | awk '{ if ($2 != "0" && $2 != "." && \
+                     $3 != "0" && $3 != "." && \
+                     $4 != "0" && $4 != ".") print $0 }' \
+        | sort -R --random-source <( yes ~{random_seed} ) \
+        > cleaned.shuffled.fam
+        awk -v max_trios="~{max_trios}" 'NR <= max_trios' cleaned.shuffled.fam \
+        | cat fam_header.txt - \
+        > cleaned.subset.fam
+        echo -e "SUBSETTED TO $( cat cleaned.subset.fam | wc -l | awk '{ print $1-1 }' ) RANDOM FAMILIES"
+      else
+        echo -e "NUMBER OF TRIOS DETECTED ( $n_trios ) LESS THAN MAX_TRIOS ( ~{max_trios} ); PROCEEDING WITHOUT DOWNSAMPLING"
+        cp cleaned.fam cleaned.subset.fam
+      fi
+      
+      # Run family analysis
+      echo -e "STARTING FAMILY-BASED ANALYSIS"
+      /opt/sv-pipeline/scripts/vcf_qc/analyze_fams.R \
+        -S /opt/sv-pipeline/scripts/vcf_qc/SV_colors.txt \
+        ~{vcf_stats} \
+        cleaned.subset.fam \
+        ~{prefix}_perSample/ \
+        ~{prefix}_perFamily_plots/
+
+    else
+
+      mkdir ~{prefix}_perFamily_plots/
+
+    fi
+
+    # Prepare output
+    echo -e "COMPRESSING RESULTS AS A TARBALL"
+    tar -czvf ~{prefix}.plotQC_perFamily.tar.gz \
+      ~{prefix}_perFamily_plots
+  >>>
+
+  output {
+    File perFamily_plots_tarball = "~{prefix}.plotQC_perFamily.tar.gz"
+    File cleaned_fam_file = "cleaned.fam"
+  }
+}
+
+
+# Plot per-sample benchmarking
+task PlotQcPerSampleBenchmarking {
+  input {
+    File per_sample_benchmarking_tarball
+    File samples_list
+    String comparison_set_name
+    String prefix
+    String sv_pipeline_qc_docker
+    Int? max_samples = 3000
```

Comment:
See previous comment about setting task defaults for optional inputs - for `max_samples` and `random_seed`